chain/network/src/peer/peer_actor.rs

```diff
@@ -332,48 +343,63 @@ impl PeerActor {
-                    tracker: Default::default(),
-                    stats,
-                    routed_message_cache: LruCache::new(
-                        NonZeroUsize::new(ROUTED_MESSAGE_CACHE_SIZE).unwrap(),
-                    ),
-                    protocol_buffers_supported: false,
-                    force_encoding,
-                    peer_info: match &stream_type {
-                        tcp::StreamType::Inbound => None,
-                        tcp::StreamType::Outbound { peer_id, .. } => Some(PeerInfo {
-                            id: peer_id.clone(),
-                            addr: Some(peer_addr),
-                            account_id: None,
-                        }),
-                    }
-                    .into(),
-                    network_state,
-                    received_messages_rate_limits,
+        // Start PeerActor on separate tokio runtime.
+        let builder = actor_system.new_tokio_builder();
+        let handle = builder.handle();
+        let stream_id = stream.id();
+        let peer_addr = stream.peer_addr;
+        let stream_type = stream.type_.clone();
+        let stats = Arc::new(connection::Stats::default());
+        let framed = stream::FramedStream::spawn(
+            handle.clone().into_sender(),
+            handle.clone().into_sender(),
+            &*handle.future_spawner(),
+            stream,
+            stats.clone(),
+        );
+        let actor = Self {
+            closing_reason: None,
+            clock,
+            handle: handle.clone(),
+            my_node_info,
+            stream_id,
+            peer_addr,
+            peer_type: match &stream_type {
+                tcp::StreamType::Inbound => PeerType::Inbound,
+                tcp::StreamType::Outbound { .. } => PeerType::Outbound,
+            },
+            peer_status: PeerStatus::Connecting(send, connecting_status),
+            framed,
+            tracker: Default::default(),
+            stats,
+            routed_message_cache: LruCache::new(
+                NonZeroUsize::new(ROUTED_MESSAGE_CACHE_SIZE).unwrap(),
+            ),
+            protocol_buffers_supported: false,
+            force_encoding,
+            peer_info: match &stream_type {
+                tcp::StreamType::Inbound => None,
+                tcp::StreamType::Outbound { peer_id, .. } => {
+                    Some(PeerInfo { id: peer_id.clone(), addr: Some(peer_addr), account_id: None })
                 }
-            }),
-            recv,
-        ))
+            }
+            .into(),
+            network_state,
+            received_messages_rate_limits,
+            registration_buffered_actions: None,
+        };
+        builder.spawn_tokio_actor(actor);
+        Ok((handle, recv))
+    }
+
+    /// If registration is in progress, queue up execution of `f` until it is complete.
+    /// Otherwise, executes `f` immediately.
+    fn delay_if_registering(&mut self, f: impl FnOnce(&mut Self) + Send + 'static) {
```

Comment:
TBH I liked the old one better, awaiting registration future before executing `f` seem more straight forward. What's the reason that makes it ugly to implement it with the new actors?